### DISCUSSION TITLE
Language server not attached to buffer

### DISCUSSION THREAD

**Discussion by suntsu42** [mason-org/mason.nvim#217] [Q&A] [answered]
Tested on linux (manjaro) using nvim (latest version of 7 and 8)

# Plugins
```lua
	Plug 'williamboman/mason.nvim'
	Plug 'williamboman/mason-lspconfig.nvim'
	Plug 'neovim/nvim-lspconfig'
```

# Mason setup
```lua
require("mason").setup()
require("mason-lspconfig").setup()
``` 

# Behavior
I'm able to install any language server without problem. 
But when i check the state using **LspInfo**, i get the following info:
```
 Language client log: /home/suntsu/.local/state/nvim/lsp.log
 Detected filetype:   cs
 
 0 client(s) attached to this buffer: 
 
 Configured servers list: 
``` 
The logfile lsp.log doesn't exist. 
I've tested it with lua and omnisharp language servers. 

The behavior remains the same when i start is manually using **LspStart**. In this case, LspStart is shown in the command bar but it doesn't start the server nor does it show any error messages. 

# Question
Do i need to do anything in order to load the language server?

**Comment by kylo252** [comment, marked as answer]
you need to call the setup function for each server manually, this plugin doesn't handle that for you

```lua
require("mason").setup()
require("mason-lspconfig").setup({ automatic_installation = { exclude = { "clangd" } } })
local server_opts = {}
require("lspconfig").omnisharp.setup(server_opts)
```

**Reply by suntsu42** [reply]
I've just seen that this is actually documented in **:h mason-lspconfig**, i just didn't understand it. 
cheers
